test/performance/performance.go

```diff
@@ -152,3 +158,69 @@ func sanitizedURL(endpoint string) string {
 	}
 	return endpoint
 }
+
+// DeploymentStatus is a struct that wraps the status of a deployment.
+type DeploymentStatus struct {
+	DesiredReplicas int32
+	ReadyReplicas   int32
+}
+
+// FetchDeploymentStatus creates a channel that can return the up-to-date DeploymentStatus periodically.
+func FetchDeploymentStatus(
+	ctx context.Context, namespace string, selector labels.Selector, ticker *time.Ticker,
+) <-chan []DeploymentStatus {
+	dl := deploymentinformer.Get(ctx).Lister()
+	ch := make(chan []DeploymentStatus)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				// Overlay the desired and ready pod counts.
+				deployments, err := dl.Deployments(namespace).List(selector)
+				if err != nil {
+					log.Printf("Error listing deployments: %v", err)
+					break
+				}
+
+				ds := make([]DeploymentStatus, len(deployments))
+				for _, d := range deployments {
+					ds = append(ds, DeploymentStatus{
+						DesiredReplicas: *d.Spec.Replicas,
+						ReadyReplicas:   d.Status.ReadyReplicas,
+					})
+				}
+				ch <- ds
```

Comment:
I think you can `select` on sending things and if the channel is closed, we can close our ticker and return

Reply:
Done.